src/util/normalizeThought.ts

```diff
@@ -49,6 +51,17 @@ const normalizeThought = moize(
 
     const strippedLowerCase = stripped.toLowerCase()
 
+    // Preserve number format, including decimal points and commas
+    if (/^-?\d+([.,]\d+)?$|^\d+,\s*\d+$/.test(strippedLowerCase)) {
+      // Return the original string to preserve the exact format
+      return s
+    }
+
+    // Special case for single letters followed by 's'
```

Comment:
FYI Instead of just saying that it is a special case (which is a given since it is a separate if statement), explain why it is needed. e.g. `// Avoid singularizing single letters followed by 's' so that 'as' does not match 'a'`